pkg/aws/core.go

```diff
@@ -185,14 +185,12 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 	if providerSpec.CapacityReservationTarget != nil {
 		if providerSpec.CapacityReservationTarget.CapacityReservationResourceGroupArn != nil {
 			inputConfig.CapacityReservationSpecification = &ec2.CapacityReservationSpecification{
-				CapacityReservationPreference: aws.String("open"),
```

Comment:
@himanshu-kun  do you want to break this down into 2 - essentially in the first iteration remove the hard coded value of open (which is anyways the default if no CapacityReservationTarget is set) and in the second iteration also introduce `CapacityReservationPreference` as part of `AWSProviderSpec.CapacityReservationTarget`?

Reply:
Yes sounds good @unmarshall.

Reply:
Sure no problem.  I have put together a version which I think handles part 2 as you describe. Do you want me to raise that as a new PR or should I push that here?

Reply:
It depends, if you want this to be merged now then we can do that and you can raise another PR. However since this commit is not going to be hot fixed, it might not make much of a difference anyways. So maybe we can add to this PR as well.

Reply:
OK I will bring it into this PR.

Reply:
thanks, kindly link PR to this issue -> https://github.com/gardener/machine-controller-manager-provider-aws/issues/116

Regarding pipeline, you will have to be part of gardener organization to take a look at it, the error because of some inner details, I'll fix and re-run it.

Reply:
Changes for part 2 are now in this PR.